src/test/java/com/commercetools/sync/products/helpers/VariantReferenceResolverTest.java

```diff
@@ -26,461 +24,418 @@
+            getMockProductService(PRODUCT_ID),
+            getMockCategoryService(CATEGORY_ID));
     }
 
     @Test
-    void resolveAssetsReferences_WithEmptyAssets_ShouldNotResolveAssets() {
-        final ProductVariantDraftBuilder productVariantDraftBuilder = ProductVariantDraftBuilder.of()
-                                                                                                .assets(emptyList());
-
-        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
-            .resolveAssetsReferences(productVariantDraftBuilder)
-            .toCompletableFuture().join();
+    void resolveReferences_WithNoAttributes_ShouldReturnEqualDraft() {
+        // preparation
+        final ProductVariantDraft productVariantDraft = ProductVariantDraftBuilder
+            .of()
+            .build();
 
-        final List<AssetDraft> resolvedBuilderAssets = resolvedBuilder.getAssets();
-        assertThat(resolvedBuilderAssets).isEmpty();
+        // test
+        final ProductVariantDraft resolvedAttributeDraft =
+            referenceResolver.resolveReferences(productVariantDraft)
+                             .toCompletableFuture()
+                             .join();
+        // assertions
+        assertThat(resolvedAttributeDraft).isEqualTo(productVariantDraft);
     }
 
     @Test
-    void resolveAssetsReferences_WithNullAssets_ShouldNotResolveAssets() {
-        final ProductVariantDraftBuilder productVariantDraftBuilder = ProductVariantDraftBuilder.of();
-
-        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
-            .resolveAssetsReferences(productVariantDraftBuilder)
-            .toCompletableFuture().join();
+    void resolveReferences_WithEmptyAttributes_ShouldReturnEqualDraft() {
+        // preparation
+        final ProductVariantDraft productVariantDraft = ProductVariantDraftBuilder
+            .of()
+            .attributes(emptyList())
+            .build();
 
-        final List<AssetDraft> resolvedBuilderAssets = resolvedBuilder.getAssets();
-        assertThat(resolvedBuilderAssets).isNull();
+        // test
+        final ProductVariantDraft resolvedAttributeDraft =
+            referenceResolver.resolveReferences(productVariantDraft)
+                             .toCompletableFuture()
+                             .join();
+        // assertions
+        assertThat(resolvedAttributeDraft).isEqualTo(productVariantDraft);
     }
 
     @Test
-    void resolveAssetsReferences_WithANullAsset_ShouldNotResolveAssets() {
-        final ProductVariantDraftBuilder productVariantDraftBuilder =
-            ProductVariantDraftBuilder.of().assets(singletonList(null));
-
-        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
-            .resolveAssetsReferences(productVariantDraftBuilder)
-            .toCompletableFuture().join();
+    void resolveReferences_WithANullAttribute_ShouldReturnDraftWithoutNullAttribute() {
```

Comment:
for this test null attribute is not added to the list, but here `resolveReferences_WithNullAttributeValue_ShouldReturnEqualDraft` it's returning the same attribute list, should it return only text attribute? 🤔

Reply:
- We remove the null sub resources (like attributes) since it doesn't make much sense to return them for the user of the method. It is also fine because the resolver doesn't mutate the actual users supplied drafts. Anyhow, I will add it as a comment for now.

- No the reference resolver shouldn't remove an attribute with a `null` value. Because it could be that the user is actually unsetting a certain attribute value which is a valid case.

Reply:
looks better now 👍